### PR TITLE
Release/v8.7.0

### PR DESCRIPTION
**Pull request by berthetclement**

### CODE REVIEW

File: R/removeBindingConstraint.R [hunk 1]
```diff
@@ -2,69 +2,88 @@
...
 #' }
-removeBindingConstraint <- function(name, opts = antaresRead::simOptions()) {
+removeBindingConstraint <- function(name = NULL, 
+                                    group = NULL, 
+                                    opts = antaresRead::simOptions()) {
   
   assertthat::assert_that(inherits(opts, "simOptions"))
   
+  # some checks for "group" parameter according to study version
+  if(!opts$antaresVersion >= 870 & !is.null(group))
```
Comment (Nekmek7) [inline R/removeBindingConstraint.R:47]: Code alternatif qui testerait moins de conditions ?
```r
  if(!is.null(group)){
    if(!opts$antaresVersion >= 870){
      stop("Parameter 'group' is only for Antares study version >= v8.7.0", 
           call. = FALSE)
    }
    else if(!is.null(name)){
      stop("You can only delete binding constraint by id/name or by group", 
           call. = FALSE)
    }
  }
```

Comment (berthetclement) [reply R/removeBindingConstraint.R]: this code is now merged

File: R/scenarioBuilder.R [hunk 2]
```diff
@@ -92,65 +108,160 @@
...
 #' }
-scenarioBuilder <- function(n_scenario, 
+scenarioBuilder <- function(n_scenario = 1, 
                             n_mc = NULL,
                             areas = NULL,
                             areas_rand = NULL,
+                            group_bc = NULL, 
+                            group_bc_rand = NULL,
                             coef_hydro_levels = NULL,
+                            mode = NULL,
```
[the remark this answers is not in the document]
Comment (berthetclement) [reply R/scenarioBuilder.R:118]: Yes it's needed if you call function in default mode, see tests below

[the remark this answers is not in the document]
Comment (KKamel67) [reply R/scenarioBuilder.R:118]: Question is : presence of argument group_bc or/and group_bc_rand can tell that you are in bc mode? So you don't need this argument in the function anymore

File: R/scenarioBuilder.R [hunk 3]
```diff
@@ -170,29 +281,33 @@ create_referential_series_type <- function(){
 readScenarioBuilder <- function(ruleset = "Default Ruleset",
                                 as_matrix = TRUE,
                                 opts = antaresRead::simOptions()) {
-  assertthat::assert_that(inherits(opts, "simOptions"))
+assertthat::assert_that(inherits(opts, "simOptions"))
```
[the remark this answers is not in the document]
Comment (berthetclement) [reply R/scenarioBuilder.R:284]: legacy

File: R/scenarioBuilder.R [hunk 4]
```diff
@@ -92,65 +108,160 @@
...
+  # check version >=870 from group parameter
+  if(!opts$antaresVersion >= 870 & !is.null(group_bc))
+    stop("Parameter 'group_bc' is only for Antares study version >= v8.7.0", 
+         call. = FALSE)
+  if(!opts$antaresVersion >= 870 & !is.null(group_bc_rand))
+    stop("Parameter 'group_bc_rand' is only for Antares study version >= v8.7.0", 
+         call. = FALSE)
+  
+  # >=v870
+  if(opts$antaresVersion >= 870){
```
[the remark this answers is not in the document]
Comment (KKamel67) [reply R/scenarioBuilder.R:139]: `	is_compatible_bc <- is_870 && !is.null(mode) && mode %in% "bc"
  if(is_compatible_bc){
    .manage_parameter_bc(n_scenario = n_scenario,
                         n_mc = n_mc,
                         group_bc = group_bc, 
                         group_bc_rand = group_bc_rand, 
                         opts = opts)
	} else {
    .manage_parameter(n_scenario = n_scenario,
                      n_mc = n_mc,
                      areas = areas,
                      areas_rand = areas_rand,
                      coef_hydro_levels = coef_hydro_levels,
                      opts = opts)
  }`

File: R/scenarioBuilder.R [hunk 5]
```diff
@@ -92,65 +108,160 @@
...
+    .manage_parameter(n_scenario = n_scenario,
+                      n_mc = n_mc,
+                      areas = areas,
+                      areas_rand = areas_rand,
+                      coef_hydro_levels = coef_hydro_levels,
+                      opts = opts)
+}
+
+# Private function of scenarioBuilder()
+  # <v870 paradigm
```
[the remark this answers is not in the document]
Comment (KKamel67) [reply R/scenarioBuilder.R:165]: Commentaire .manage_parameter: # <v870 paradigm --> # <v870 paradigm ou >=v870 sans contrainte couplante.

En se référant à l'appel de cette fonction.

File: R/scenarioBuilder.R [hunk 6]
```diff
@@ -170,29 +281,33 @@ create_referential_series_type <- function(){
...
-  assertthat::assert_that(inherits(opts, "simOptions"))
+assertthat::assert_that(inherits(opts, "simOptions"))
+  
+  # read existing scenariobuilder.dat
   if (is_api_study(opts)) {
     if (is_api_mocked(opts)) {
       sb <- list("Default Ruleset" = NULL)
     } else {
-      sb <- readIni("settings/scenariobuilder", opts = opts, default_ext = ".dat")
+      sb <- readIni("settings/scenariobuilder",
```
[the remark this answers is not in the document]
Comment (KKamel67) [reply R/scenarioBuilder.R:292]: It seems that it is legacy code

**Review by vargastat — COMMENTED** [review]
ready for merge into develop

**Pull request merged**